packages/synthetix-main/contracts/storage/Distribution.sol

```diff
@@ -89,151 +137,137 @@ library Distribution {
-
     /**
-     * @dev Adds or removes value to the distribution. The value is
-     * distributed into each individual share by altering the distribution's `valuePerShare`.
+     * @dev Inflates or deflates the total value of the distribution by the given value.
+     *
+     * The value being distributed ultimately modifies the distribution's valuePerShare.
      */
-    function distributeValue(Data storage dist, int amount) internal {
-        if (amount == 0) {
+    function distributeValue(Data storage dist, int value) internal {
+        if (value == 0) {
             return;
         }
 
-        uint totalShares = dist.totalShares;
+        uint totalShares = dist.totalShares.uint128toUint256();
 
         if (totalShares == 0) {
             revert EmptyDistribution();
         }
 
         // TODO: Can we safely assume that amount will always be a regular integer,
         // i.e. not a decimal?
-        int amountHighPrecision = amount.toHighPrecisionDecimal();
-        int deltaValuePerShare = amountHighPrecision / int(totalShares);
+        int valueHighPrecision = value.toHighPrecisionDecimal();
+        int deltaValuePerShare = valueHighPrecision / int(totalShares);
 
         dist.valuePerShare += int128(deltaValuePerShare);
     }
 
     /**
      * @dev Updates an actor's number of shares in the distribution to the specified amount.
      *
-     * Modifies the distribution's totalValue, since the number of shares is changed.
-     *
-     * Eg. if this is used to increase an actor's number of shares, the distribution's total value will increase
-     * by deltaShares * valuePerShare.
-     *
-     * Returns the actor's individual change in value.
-     *
-     * Note: A distribution is intended to either use updateActorShares or updateActorValue, but not both
-     * during the distribution's lifetime.
+     * Whenever an actor's shares are changed in this way, we record the distribution's current valuePerShare into the actor's lastValuePerShare record.
      */
-    function updateActorShares(
+    function setActorShares(
         Data storage dist,
         bytes32 actorId,
-        uint shares
-    ) internal returns (int changedValue) {
+        uint newActorShares
+    ) internal {
         DistributionActor.Data storage actor = dist.actorInfo[actorId];
 
-        // Compare the current valuePerShare to the valuePerShare at
-        // the last time the actor's shares were updated.
-        int128 deltaValuePerShare = dist.valuePerShare - actor.lastValuePerShare;
-
-        // Calculate the total change in the actor's value.
-        int changedValueHighPrecision = deltaValuePerShare * actor.shares.uint128toInt256();
-        changedValue = changedValueHighPrecision.fromHighPrecisionDecimalToInteger();
-
-        // Modify the total shares with the actor's change in shares.
-        uint128 sharesUint128 = shares.uint256toUint128();
```

Comment:
This is overloading. Moved it to a new `getActorValueChange()` function, and used the new function externally where appropriate.